nifi-extension-bundles/nifi-standard-services/nifi-oauth2-provider-bundle/nifi-oauth2-provider-service/src/main/java/org/apache/nifi/oauth2/StandardOauth2AccessTokenProvider.java

```diff
@@ -339,6 +340,31 @@ public AccessToken getAccessDetails() {
         return accessDetails;
     }
 
+    @Override
+    public void refreshAccessDetails() {
+
+        if (this.accessDetails == null || this.accessDetails.getRefreshToken() == null) {
+            acquireAccessDetails();
+            return;
+        }
+
+        getLogger().debug("Refresh Access Token request started [{}]", authorizationServerUrl);
+
+        FormBody.Builder refreshTokenBuilder = new FormBody.Builder()
+                .add("grant_type", "refresh_token")
+                .add("refresh_token", this.accessDetails.getRefreshToken());
+
+        addFormData(refreshTokenBuilder);
+
+        AccessToken newAccessDetails = requestToken(refreshTokenBuilder);
+
+        if (newAccessDetails.getRefreshToken() == null) {
+            newAccessDetails.setRefreshToken(this.accessDetails.getRefreshToken());
+        }
+
+        this.accessDetails = newAccessDetails;
```

Comment:
Minor, but I recommend placing the logic in an `else` condition instead of the short-circuit return.